syntax = "proto3";
package steward;

import "proto/steward/organization.proto";

message User {
  string _id = 1;
  string name = 2;
  string password = 3;
  string email = 4;
  float available_effort = 5;
  string organization_id = 6;
}

enum UserLevel {
  ANONYMOUS = 0;
  USER = 1;
  OWNER = 2;
}

message ACL {
  string _id = 1;
  User user = 2;
  Organization organization = 3;
  UserLevel level = 4;
}

message GetUserRequest {
  string _id = 1;
  string email = 2;
}

message DeleteUserRequest {
  string _id = 1;
}

message UpdateUserRequest {
  string _id = 1;
  User user = 2;
}

message ListUsersRequest {
}

message CreateUserRequest {
  string name = 1;
  string password = 2;
  string email = 3;
  float available_effort = 4;
  string organization_id = 5;
}
